incubator/hnc/test/e2e/demo_test.go

```diff
@@ -93,8 +93,91 @@ var _ = Describe("Demo", func() {
 		RunShouldNotContain("my-creds", 2, "kubectl -n", nsService1, "get secrets")
 	})
 
-	It("Should create and delete subnamespaces", func() {
-		// set up initial structure
+	It("Should intergrate hierarchical network policy", func(){
+		MustRun("kubectl create ns", nsOrg)
+		MustRun("kubectl hns create", nsTeamA, "-n", nsOrg)
+		MustRun("kubectl hns create", nsTeamB, "-n", nsOrg)
+		MustRun("kubectl hns create", nsService1, "-n", nsTeamA)
+		MustRun("kubectl hns create", nsService2, "-n", nsTeamA)
+		// create a web service s2 in namespace service-2, and a client pod client-s1 in namespace service-1 that can access this web service
+		MustRun("kubectl run s2 -n", nsService2, "--image=nginx --restart=Never --expose --port 80")
+		// at least 20 seconds is needed here from experiments 
+		RunShouldContain("Welcome to nginx!", 20, 
+			"kubectl run client -n", nsService1, "-i --image=alpine --restart=Never --rm -- sh -c", "\"wget -qO- --timeout 2 http://s2.service-2\"")
+		RunShouldContain("Welcome to nginx!", 10, 
+			"kubectl run client -n", nsTeamA, "-i --image=alpine --restart=Never --rm -- sh -c", "\"wget -qO- --timeout 2 http://s2.service-2\"")
+		RunShouldContain("Welcome to nginx!", 10, 
+			"kubectl run client -n", nsTeamB, "-i --image=alpine --restart=Never --rm -- sh -c", "\"wget -qO- --timeout 2 http://s2.service-2\"")
+
+		// create a default network policy that blocks any ingress from other namespaces 
+		policy := "" +
+			"kind: NetworkPolicy" + "\n" +
+			"apiVersion: networking.k8s.io/v1" + "\n" +
+			"metadata:" + "\n" +
+			"  name: deny-from-other-namespaces" + "\n" +
+			"  namespace: acme-org" + "\n" +
+			"spec:" + "\n" +
+			"  podSelector:" + "\n" +
+			"    matchLabels:" + "\n" +
+			"  ingress:" + "\n" +
+			"  - from:" + "\n" +
+			"    - podSelector: {}"
+		filename := "temp.yaml"
+		WriteToFile(policy, filename)
```

Comment:
Suggestion: this should be something like:

```
tmp := WriteTempFile(policy)
defer RemoveTempFile(tmp)
MustRun("kubectl apply -f", tmp)
```

The file should be created in the `/tmp` directory

Reply:
Also _consider_ https://golang.org/pkg/io/ioutil/#TempFile but don't feel obligated to use.

Reply:
I tried this approach and kept getting 
```  Expected
      <*os.PathError | 0xc0002ad3b0>: {
          Op: "remove",
          Path: "tempfile.yaml",
          Err: 0x2,
      }
  to be nil```

Reply:
Did you create two `tempfile.yaml` copies and try to delete them twice?

Reply:
done